### ISSUE TITLE
Update elasticsearch_logging_discovery to use v1 API

### ISSUE THREAD

**Issue by nikhiljindal** [GoogleCloudPlatform/kubernetes#10756]
I see that elasticsearch_logging_discovery is still issuing v1beta3 requests in e2e tests.
We need to update it to use v1 API.

We cant stop exposing v1beta3 in master, unless this is fixed (#10525)

Example request from e2e logs:
```
GET /api/v1beta3/namespaces/default/services/elasticsearch-logging: (260.054µs) 404 [[elasticsearch_logging_discovery/v0.18.0 (linux/amd64) kubernetes/      unknown] 10.240.92.0:37293]
```

**Comment by nikhiljindal** [comment]
cc @satnam6502 @lavalamp @bgrant0607 
Any idea on how to fix it?
Will building a new image from HEAD fix this?

**Comment by satnam6502** [comment]
Yes, indeed, there don't seem to be any hard-coded references to `v1beta3` so the unversioned client library in our Go code is probably the issue. I will rebuild from HEAD to see if this fixes the issue.

**Comment by nikhiljindal** [comment]
Thanks @satnam6502

**Comment by lavalamp** [comment]
Rebuilding is probably sufficient.

**Comment by satnam6502** [comment]
I am creating a new tag and updating -- out of paranoia -- to avoid stomping over an existing image.

**Comment by wojtek-t** [comment]
Fixed by #10801
